### DISCUSSION TITLE
Scripts a la carte: no examples on using input parameters

### DISCUSSION THREAD

**Discussion by nick-youngblut** [nextflow-io/nextflow#3527] [Q&A]
In the [Nextflow docs](https://www.nextflow.io/docs/edge/process.html#scripts-a-la-carte), there are no examples of using input parameters for perl, python, R, etc. So, it is unclear how one would process input files, as done for bash script examples (where inputs are encoded as `$MY_INPUT`). It would be helpful to incorporate this into into the docs.

**Comment by mribeirodantas** [comment]
Hello, @nick-youngblut. I'm not sure if I understand your suggestion. Shell script is a scripting language just like Python or Perl. The way Nextflow interacts with them is just the same.

```
process PYTHON {
  input:
    val x
  output:
    stdout
  script:
  """
  #!/usr/bin/env python

  print("$x")
  """
}

process SHELL {
  input:
    val x
  output:
    stdout
  script:
  """
  echo $x
  """
}

process PERL {
  input:
    val x
  output:
    stdout
  script:
  """
  #!/usr/bin/env perl

  print("$x")
  """
}

workflow {
  Channel
    .of("Hello", "World")
    .set { my_ch }
  PYTHON(my_ch)
  PERL(my_ch)
  SHELL(my_ch)
}
```

You're suggesting us to say so explicitly? To write more examples on other scripting languages?

**Reply by nick-youngblut** [reply]
Currently, the examples:

```
process perlStuff {
    """
    #!/usr/bin/perl

    print 'Hi there!' . '\n';
    """
}

process pythonStuff {
    """
    #!/usr/bin/python

    x = 'Hello'
    y = 'world!'
    print "%s - %s" % (x,y)
    """
}
```

do not show that feature. They could just be modified slightly to be more explicit, as you demonstrate: 

```
process perlStuff {
    input:
    val(x)

    """
    #!/usr/bin/perl

    print 'Hi there!' . '\n';
    print("$x")
    """
}

process pythonStuff {
    input:
    val(x)
    path(y)

    """
    #!/usr/bin/python

    print('Your value: {}'.format($x))
    with open($y) as inF:
        for line in inF:
           print(line)
    """
}
```

I could send a PR, if you are OK with these updates (not sure if the python code that I wrote works correctly).

**Reply by mribeirodantas** [reply]
Submitting a PR would be great, @nick-youngblut !

If you're going to do it, I would like to suggest you to also add the input/output blocks to these examples. They're not in the example in the documentation currently and I fear they may lead the reader to an incorrect understanding of the usage of different scripting languages with Nextflow.
